### PR TITLE
Fix Dialog to have `--salt-zIndex-modal`

### PR DESCRIPTION
**Pull request by origami-z**
Dialog to have modal z-index.
Scrim remains as Drawer zindex - 1, given Drawer also uses it.

### CODE REVIEW

File: site/propsGen.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 const docgen = require("react-docgen-typescript");
```
Comment (joshwooding) [inline site/propsGen.js:1]: Were these changes intentional?

Comment (origami-z) [reply site/propsGen.js:1]: nope... i'll remove

**Review by joshwooding — APPROVED** [review]
